Source/Project64/N64 System/Mips/Pif Ram.cpp

```diff
@@ -297,19 +297,16 @@ void CPifRam::SI_DMA_READ()
 			{
 				continue;
 			}
-			RDRAM[RdramPos ^3] = m_PifRam[count];
+			RDRAM[BES(RdramPos)] = m_PifRam[count];
 		}
 	}
 	else
 	{
-		for (size_t i = 0; i < 64; i += 4)
-		{
-			unsigned __int32 pif_ram_dword;
-			std::memcpy(&pif_ram_dword, &PifRamPos[i], sizeof(unsigned __int32));
-
-			pif_ram_dword = swap32by8(pif_ram_dword);
+		register size_t i;

```

Comment:
>Sure, but this isn't enforced in C++ (in C, go nuts if you like).

OK.  Well the reason why that's unfortunate, is because Project64 is written in C++. :)

But that means I should take your advice to heart, so I'll remove the "register" in a bit.

>If you replied it was deprecated, and it was indeed the case (I would look for sources first to back it up or ask you to provide it), I would move code over to the non-deprecated form once confirmed; 

Would you, now?  I almost find that impressive.

So if I'm reading what you said right (and assuming you read my post right):  if this crazy hypothetical future spec would deprecate the `for (type i =` style, you would adjust by moving over to the non-deprecated C89-ish counter style simply because it isn't deprecated.

That means that in spite of all these scope security and readability defenses for the C99 style you have cited, ultimately the deciding factor for you is what does the modernistic, newer spec have deprecated/not deprecated.

Well, no insult, I can genuinely applaud you for being so tolerant of basing correct "software maintenance" as such a simple matter of what other people (namely these official "committees") think.  It isn't that I can't respect this, but I personally see no point to being controlled singularly by the opinions of others--except of course when you are working for a company or other's project that persists the use of such a spec, then it's a no-brainer.  But for your own code and your own projects?  You could just as easily set the saner compiler to use an older version of the spec.

Like I said:  with "deprecation" always comes a reason, and that reasoning is all that counts.
Especially if they deprecate something that contradicts what they added to C99.

>basic software maintenance 101.

Good theory but I have an even easier one:  Respect backwards compatibility.

Then there is no cause for having to mass-change/maintain updates like that.

>literally_any_newer_language_spec.txt.

If they disregard backwards compat., then yes.  (See also:  OpenGL deprecation)

>But it is and can be. Scoping rules exist.

And always have existed--just as they have in C89.  However, whether the loop heading (as opposed to the body) constitutes its own existing "scope" is again a matter of one's own (and the individual language's) personal interpretation.  Maybe that could be another reason why ANSI C does not support what you say.

>You don't need to concern yourself with the type.

If the type is irrelevant, then I would think that to be one more reason why `for (i = 0; i < N; i++)` is easier to read than `for (arbitrary_type_here i = 0; i < N; i++)`.

Never did recall seeing "int" or "size_t" in any Calculus textbooks about sequences.
Hmm...could that be because the type is irrelevant?

>You can obsolete the need to change it by using auto in many circumstances where this would actually matter in practice, [...]

Possibly, but in my example `i` is also the parameter to 3 different functions.  For whatever reason (optimization, problems with porting, or even fixing design problems like you said--maybe my example is too "unique") it just is faster to debug/analyze to switch back-and-forth between by changing one central declaration, not 3+ declarations.  Otherwise, you could use a custom typedef (like the semi-custom ones in C99's stdint.h).

>Yeah, they're what maintain and write the code, it's a pretty nice one-sided relationship :^).

Kudos to them, then.  I am glad it's not one-sided for me, at least.

I have had few, temporary portability problems when compiling C89 code on modern C/C++ compilers.  (I do not always care to do things like write "register" though.)